wrappers/core18.go

```diff
@@ -157,6 +196,10 @@ func AddSnapdSnapServices(s *snap.Info, opts *AddSnapdSnapServicesOptions, inter
 		opts = &AddSnapdSnapServicesOptions{}
 	}
 
+	if err := writeSnapdDesktopFiles(s); err != nil {
+		return err
+	}
+
```

Comment:
Maybe consider moving this to the end of the function with the other `writeSnapd*OnCore()` calls?